model/src/main/kotlin/utils/Snippet.kt

```diff
@@ -49,5 +51,12 @@ data class Snippet(
     /**
      * The license of the component the code snippet is commit from.
      */
-    val licenses: SpdxExpression
+    val licenses: SpdxExpression,
+
+    /**
+     * A map for scanner specific snippet data that cannot be mapped into any generalized property, but still needs to
+     * be made available in the scan Summary.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val additionalData: Map<String, String> = emptyMap()
```

Comment:
It just occurred to me that we already have `additionalData` in `ScanResult`. So do we really need this at this place?